wagtail/wagtailadmin/views/pages.py

```diff
@@ -434,6 +439,10 @@ def edit(request, page_id):
         else:
             if page.locked:
                 messages.error(request, _("The page could not be saved as it is locked"))
+
+            elif form.non_field_errors:
+                pass
```

Comment:
Same thing here. Also need to find relevant place for this same functionality for snippets.